bookkeeper-stats-providers/codahale-metrics-provider/src/main/java/org/apache/bookkeeper/stats/CodahaleStatsLogger.java

```diff
@@ -87,6 +87,11 @@ public T getValue() {
     }
 
     @Override
+    public <T extends Number> void unregisterGauge(String statName, Gauge<T> gauge) {
+        // do nothing right now as the Codahale doesn't support conditional removal
```

Comment:
Seems metric name is the id for each metric in Codahale, Would a "metrics.remove(metricName)" achieve this unregisterGauge?